File: frontend/src/api/TaskService.ts
import Axios from "axios"

class TaskService {
    static instance = Axios.create({
        baseURL: "http://localhost:8082/task"
    })

    public static async insertOne(task: Task) {
        let response = await this.instance.post("/", task);
        return response.data
    }

    public static async deleteOne(id: number) {
        let response = await this.instance.delete("/", {
            params: {
                id
            }
        });

        return response.data
    }

    public static async updateOne(task: Task) {
        let response = await this.instance.put("/", task);
        return response.data
    }

    public static async findAll(listid: number) {
        let response = await this.instance.get("/", {
            params: {
                listid
            }
        })

        return response.data
    }

    public static async findOne(id: number) {
        let response = await this.instance.get(`/${id}`)
        return response.data
    }
}

export default TaskService;

File: backend/src/main/java/com/example/backend/service/TaskService.java
package com.example.backend.service;

import com.example.backend.model.Task;
import com.example.backend.repository.FileItemRepository;
import com.example.backend.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class TaskService {
    @Autowired
    TaskRepository taskRepository;
    @Autowired
    FileItemService fileItemService;

    public Task insertOne(Task task) {
        return taskRepository.save(task);
    }

    public void deleteOne(Long id) {
        taskRepository.deleteById(id);
        fileItemService.findAll(id).forEach(fileItem -> {
            fileItemService.deleteOne(fileItem.getId());
        });
    }

    public Task updateOne(Task task) {
        return taskRepository.save(task);
    }

    public List<Task> findAll(Long listid) {
        return taskRepository.findAllByListid(listid);
    }

    public Optional<Task> findOne(Long id) {
        return taskRepository.findById(id);
    }
}
